src/mui/input/EmbeddedArrayInput.js

```diff
@@ -0,0 +1,161 @@
+import FlatButton from 'material-ui/FlatButton';
+import React, { Component } from 'react';
+import { FieldArray, Field } from 'redux-form';
+import PropTypes from 'prop-types';
+
+import ContentCreateIcon from 'material-ui/svg-icons/content/create';
+import ActionDeleteIcon from 'material-ui/svg-icons/action/delete';
+
+/**
+ * An Input component for generating/editing an embedded array
+ *
+ *
+ * Use it with any set of input componentents as children, like `<TextInput>`,
+ * `<SelectInput>`, `<RadioButtonGroupInput>` ... etc.
+ *
+ * @example
+ * export const CommentEdit = (props) => (
+ *     <Edit {...props}>
+ *         <SimpleForm>
+ *              <EmbeddedArrayInput source="links">
+ *                  <SimpleList>
+ *                       <TextField source="url" />
+ *                       <TextField source="context"/>
+ *                   </SimpleList>
+ *               </EmbeddedArrayInput>
+ *         </SimpleForm>
+ *     </Edit>
+ * );
+ */
+export class EmbeddedArrayInput extends Component {
+    constructor(props) {
+        super(props);
+        // nothing to do for now
+    }
+
+    renderList = ({ fields }) => {
+        const { children, arrayElStyle } = this.props;
+        return (
+            <div className="EmbeddedArrayInputContainer" style={{ margin: '1em' }}>
+                <div>
+                    {
+                        fields.map((member, index) =>
+                            <ArrayElement key={index} fields={fields} inputs={children} elStyle={arrayElStyle} member={member} index={index} />
+                        )
+                    }
+                </div>
+                <FlatButton primary style={{ float: 'right' }} icon={<ContentCreateIcon />} label="Add" onClick={() => fields.push({})} />
+                <div style={{ clear: 'both' }} />
+            </div>
+        )
+    }
+
+
+    render() {
+        const { input, resource, label, source, allowEmpty, basePath, onChange, children, meta } = this.props;
+
+        return (
+            <FieldArray name={source} component={this.renderList} />
+        )
+    }
+}
+
+const ArrayElement = ({ fields, inputs, member, index, elStyle }) => {
+    const removeElement = () => fields.remove(index);
+    const containerStyle = { padding: '0 1em 1em 1em' };
+    return (
+        <div style={elStyle} className="EmbeddedArrayInputItemContainer">
+            <div style={containerStyle}>
+                {
+                    React.Children.map(inputs, input => input && (
+                        <div key={input.props.source} className={`aor-input-${input.props.source}`} style={input.props.style}>
+                            <ArrayElementFormField input={input} prefix={member} />
+                        </div>
+                    ))
+                }
+            </div>
+            <FlatButton
```

Comment:
After Edit this is the current style (BTW, I've added separators)

![screenshot-localhost-8080-2017-06-29-15-28-23](https://user-images.githubusercontent.com/730201/27687483-b02c88fc-5cdf-11e7-9558-b42e88edea39.png)

I tried that in a `Paper` element (but didn't commit) it appeared ok too, what do you think ?
![screenshot-localhost-8080-2017-06-29-14-37-18-with-paper](https://user-images.githubusercontent.com/730201/27687517-d16f15b6-5cdf-11e7-953e-5b76e4c9ddc3.png)